types/seamless-immutable/index.d.ts

```diff
@@ -84,18 +88,75 @@ declare namespace SeamlessImmutable {
 
         replace<S>(valueObj: S, options?: ReplaceConfig): Immutable<S>;
     }
+    type ImmutableObject<T> = ImmutableObjectMixin<T> & { readonly [P in keyof T]: Immutable<T[P]> };
+
+    /** An ImmutableArray provides read-only access to the array elements, and provides functions (such as `map()`) that return immutable data structures. */
+    type ImmutableArray<T> = Readonly<ImmutableArray.Remaining<T> & ImmutableArray.Additions<T> & ImmutableArray.Overrides<T>>;
+    namespace ImmutableArray {
+        /** New methods added by seamless-immutable. */
+        interface Additions<T> {
+            asMutable(opts?: AsMutableOptions): T[];
+            asObject(toKeyValue: (item: T) => [string, any]): Immutable<object>;
```

Comment:
It might be worth adding a generic on this method, so there's more type safety building the return array

```suggestion
            asObject<U = object>(toKeyValue: (item: T) => [keyof U, any]): Immutable<U>;
```

Reply:
Thanks for the suggestion! I tweaked the change you've suggested to make the code even more type-safe by ensuring the 'value' part in the return array isn't just `any`.

``` typescript
asObject<U extends object = {}, K extends keyof U = keyof U>(
    toKeyValue: (item: T) => [K, U[K]]
): Immutable<U>;
```

Reply:
Even better!